sdk/core/core-rest-pipeline/src/retryStrategies/throttlingRetryStrategy.ts

```diff
@@ -2,54 +2,79 @@
 import { RetryStrategy } from "./retryStrategy";
 
 /**
- * Returns the number of milliseconds to wait based on a Retry-After header value.
- * Returns undefined if there is no valid value.
- * @param headerValue - An HTTP Retry-After header value.
+ * The header that comes back from Azure services representing
+ * the amount of time (minimum) to wait to retry (in seconds or timestamp after which we can retry).
  */
-function parseRetryAfterHeader(headerValue: string): number | undefined {
+const RetryAfterHeader = "Retry-After";
+/**
+ * The headers that come back from Azure services representing
+ * the amount of time (minimum) to wait to retry.
+ *
+ * "retry-after-ms", "x-ms-retry-after-ms" ==> milliseconds
+ * "Retry-After" ==> seconds or timestamp
+ */
+const AllRetryAfterHeaders: string[] = ["retry-after-ms", "x-ms-retry-after-ms", RetryAfterHeader];
+
+/**
+ * A response is a throttling retry response if it has a throttling status code (429 or 503),
+ * as long as one of the [ "Retry-After" or "retry-after-ms" or "x-ms-retry-after-ms" ] headers has a valid value.
+ *
+ * Returns the `retryAfterInMs` value if the response is a throttling retry response.
+ * If not throttling retry response, returns `undefined`.
+ *
+ * @internal
+ */
+function getRetryAfterInMs(response?: PipelineResponse): number | undefined {
+  if (!(response && [429, 503].includes(response.status))) return undefined;
   try {
-    const retryAfterInSeconds = Number(headerValue);
-    if (!Number.isNaN(retryAfterInSeconds)) {
-      return retryAfterInSeconds * 1000;
-    } else {
-      // It might be formatted as a date instead of a number of seconds
+    // Headers: "retry-after-ms", "x-ms-retry-after-ms", "Retry-After"
+    for (const header of AllRetryAfterHeaders) {
+      const retryAfterValue = parseHeaderValueAsNumber(response, header);
+      if (retryAfterValue === 0 || retryAfterValue) {
+        // "Retry-After" header ==> seconds
+        // "retry-after-ms", "x-ms-retry-after-ms" headers ==> milli-seconds
+        const multiplyingFactor = header === RetryAfterHeader ? 1000 : 1;
+        return retryAfterValue * multiplyingFactor; // in milli-seconds
+      }
+    }
 
-      const now: number = Date.now();
-      const date: number = Date.parse(headerValue);
-      const diff = date - now;
+    // RetryAfterHeader ("Retry-After") has a special case where it might be formatted as a date instead of a number of seconds
+    const retryAfterHeader = response.headers.get(RetryAfterHeader);
+    if (!retryAfterHeader) return;
 
-      return Number.isNaN(diff) ? undefined : diff;
-    }
+    const retryAfterInMS = undefined;
+    const now: number = Date.now();
+    const date: number = Date.parse(retryAfterHeader);
+    const diff = date - now;
+    if (Number.isFinite(diff)) return Math.max(0, diff); // negative diff would mean a date in the past, so retry asap with 0 milliseconds
+    return retryAfterInMS;
```

Comment:
Minor tweak, since `retryAfterInMS` is just a constant `undefined`:

```suggestion
    const now: number = Date.now();
    const date: number = Date.parse(retryAfterHeader);
    const diff = date - now;
    if (Number.isFinite(diff)) return Math.max(0, diff); // negative diff would mean a date in the past, so retry asap with 0 milliseconds
    return undefined;
```

Or if you prefer:

```suggestion
    const date = Date.parse(retryAfterHeader);
    const diff = date - Date.now();
    // negative diff would mean a date in the past, so retry asap with 0 milliseconds
    return Number.isFinite(diff) ? Math.max(0, diff) : undefined; 
```